secure/src/handshake/mod.rs

```diff
@@ -526,12 +526,22 @@ impl State {
         if self.position != Position::Hello {
             return Err(ErrorSecure::StateNotHello);
         }
-        if let Some(ck) = &mut self.cipher_creator {
-            self.shared_key = Some(ck.unpack_from_cipher(&cipher, &self.handshake_data)?);
-            return Ok(());
-        }
+        let Some(ck) = &mut self.cipher_creator else {
+            return Err(ErrorSecure::NoCipherCreator);
+        };
+        self.position = Position::SharedKey;
+        self.shared_key = Some(ck.unpack_from_cipher(&cipher, &self.handshake_data)?);
+        Ok(())
+    }
 
-        Err(ErrorSecure::NoCipherCreator)
+    pub fn get_shared_key(&self) -> Result<Vec<u8>, ErrorSecure> {
+        if self.position != Position::SharedKey {
+            return Err(ErrorSecure::StateNotSharedKey);
+        }
+        match self.shared_key.as_ref() {
+            None => return Err(ErrorSecure::StateNotSharedKey),
+            Some(sk) => Ok(sk.clone()),
```

Comment:
Question: What do we need a secret_key (shared_key) to be shared?

Isn't the point of private key to be one and only, and kept in secret?

Reply:
This is a Diffie-Hellman-like key exchange, the 3-way D-H version (faster with the least data exchanged, this is for clarity).
In that case, you don't want to use asymmetric cryptography for the encryption-decryption of your connection as it is computationally expensive. What is cheaper, and safe (even in post-quantum cryptography) is to use AES. For that thing we are creating and encapsulating the shared key (AES is symmetric cryptography).